### DISCUSSION TITLE
fresh Spack install gcc@anyversion UnsupportedMicroarchitecture error

### DISCUSSION THREAD

**Discussion by simoccc** [spack/spack#44039] [Q&A]
Hi , 

I'm installing a new fresh Spack on a Oracle Linux 8 server .

it seems to generally work, but I cannot install any new version of gcc inside Spack.I'm just following the tutorial ., I started with gcc@12 and tried older version but nothing changes.

Oracle Linux 8 comes with gcc version 8.5 I tried to install gcc@10 11 12 13 without success I always got the error : 

==> Installing gcc-10.5.0-to7v4zd2ocvdcqrtpgnbiwzrn5auxsye [26/26]
==> No binary for gcc-10.5.0-to7v4zd2ocvdcqrtpgnbiwzrn5auxsye found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/25/25109543fdf46f397c347b5d8b7a2c7e5694a5a51cce4b9c6e1ea8a71ca307c1.tar.xz
==> Applied patch /home/myuser/spack/var/spack/repos/builtin/packages/gcc/zstd.patch
==> Error: TypeError: expected string or bytes-like object

/home/itadm/spack/var/spack/repos/builtin/packages/gcc/package.py:691, in get_common_target_flags:
        688            try:
        689                return arch.optimization_flags("gcc", spec.version)
        690            except UnsupportedMicroarchitecture:
  >>    691                pass
        692        # no arch specific flags in common, unlikely to happen.
        693        return ""

I tried to clean everything or reinstall everything in another user home but nothing changes , any help?

Best Regards,
Simo

**Comment by jack-morrison** [comment]
Hi @simoccc, it looks like this is likely fixed by https://github.com/spack/spack/pull/44037.

**Reply by simoccc** [reply]
I did a git pull and thank you very much now gcc is working !!! 

Now I have an issue installing grads with %gcc@12 , could it be similar ? 

the install line is: 

spack install grads %gcc@12 +hdf4 +hdf5 +netcdf +grib2

-----

[+] /home/user/spack/opt/spack/linux-oracle8-skylake_avx512/gcc-12.3.0/cairo-1.16.0-qnlhl52habkt546rh73ezjgcii7kruzw
==> Installing grads-2.2.3-jl3opkpkqvfsptquvh63hwgmbzdscyqc [109/109]
==> No binary for grads-2.2.3-jl3opkpkqvfsptquvh63hwgmbzdscyqc found: installing from source
Process Process-54:
Traceback (most recent call last):
  File "/home/user/spack/lib/spack/spack/spec.py", line 4189, in __getitem__
    (x for x in order() if (not x.virtual) and x.package.provides(name)),
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/spack/lib/spack/spack/build_environment.py", line 1149, in _setup_pkg_and_run
    pkg, dirty=kwargs.get("dirty", False), context=Context.from_string(context)
  File "/home/user/spack/lib/spack/spack/build_environment.py", line 799, in setup_package
    env_mods.extend(setup_context.get_env_modifications())
  File "/home/user/spack/lib/spack/spack/build_environment.py", line 1060, in get_env_modifications
    spack.builder.create(pkg).setup_build_environment(env)
  File "/home/user/spack/lib/spack/spack/builder.py", line 199, in setup_build_environment
    return self.pkg_with_dispatcher.setup_build_environment(env)
  File "/home/user/spack/var/spack/repos/builtin/packages/grads/package.py", line 83, in setup_build_environment
  File "/home/user/spack/lib/spack/spack/spec.py", line 1205, in __get__
    value = f()
  File "/home/user/spack/lib/spack/spack/spec.py", line 1193, in <lambda>
    callbacks_chain.append(lambda: getattr(pkg, self.attribute_name))
  File "/home/user/spack/var/spack/repos/builtin/packages/hdf/package.py", line 133, in libs
    if "+szip" in self.spec:
  File "/home/user/spack/lib/spack/spack/spec.py", line 4193, in __getitem__
    raise KeyError(f"No spec with name {name} in {self}")
KeyError: 'No spec with name zlib in hdf@=4.2.15%gcc@=12.3.0+external-xdr~fortran~java~netcdf+pic~shared~szip build_system=autotools patches=49733dd,5434f29,ce75518,d54592d,f7d67e8 arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] berkeley-db@=18.1.40%gcc@=8.5.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-oracle8-skylake_avx512 ^[deptypes=build] bison@=3.8.2%gcc@=12.3.0~color build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] bzip2@=1.0.8%gcc@=8.5.0~debug~pic+shared build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build] ca-certificates-mozilla@=2023-05-30%gcc@=8.5.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build] cmake@=3.27.9%gcc@=12.3.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] curl@=8.7.1%gcc@=8.5.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-oracle8-skylake_avx512 ^[deptypes=build] diffutils@=3.10%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] findutils@=4.9.0%gcc@=8.5.0 build_system=autotools patches=440b954 arch=linux-oracle8-skylake_avx512 ^[deptypes=build] flex@=2.6.3%gcc@=12.3.0+lex~nls build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=link] gcc-runtime@=12.3.0%gcc@=12.3.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=link] gcc-runtime@=8.5.0%gcc@=8.5.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gdbm@=1.23%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gettext@=0.22.5%gcc@=8.5.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=iconv,libc] glibc@=2.28%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gmake@=4.4.1%gcc@=8.5.0~guile build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] krb5@=1.20.1%gcc@=12.3.0+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=jpeg] libjpeg-turbo@=3.0.0%gcc@=12.3.0~ipo~jpeg8~partial_decoder+pic build_system=cmake build_type=Release generator=make libs=shared,static arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] libsigsegv@=2.14%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=rpc] libtirpc@=1.3.3%gcc@=12.3.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] libxml2@=2.10.3%gcc@=8.5.0+pic~python+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] m4@=1.4.19%gcc@=8.5.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-oracle8-skylake_avx512 ^[deptypes=build] nasm@=2.15.05%gcc@=12.3.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] ncurses@=6.5%gcc@=8.5.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] nghttp2@=1.57.0%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] openssl@=3.3.0%gcc@=8.5.0~docs+shared build_system=generic certs=mozilla arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] perl@=5.38.0%gcc@=8.5.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-oracle8-skylake_avx512 ^[deptypes=run] pigz@=2.8%gcc@=8.5.0 build_system=makefile arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run virtuals=pkgconfig] pkgconf@=2.2.0%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] readline@=8.2%gcc@=8.5.0 build_system=autotools patches=bbf97f1 arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] tar@=1.34%gcc@=8.5.0 build_system=autotools zip=pigz arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] xz@=5.4.6%gcc@=8.5.0~pic build_system=autotools libs=shared,static arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=zlib-api] zlib-ng@=2.1.6%gcc@=8.5.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] zstd@=1.5.6%gcc@=8.5.0+programs build_system=makefile compression=none libs=shared,static arch=linux-oracle8-skylake_avx512'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/spack/lib/spack/spack/spec.py", line 4189, in __getitem__
    (x for x in order() if (not x.virtual) and x.package.provides(name)),
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/user/spack/lib/spack/spack/build_environment.py", line 1168, in _setup_pkg_and_run
    package_context = get_package_context(tb)
  File "/home/user/spack/lib/spack/spack/build_environment.py", line 1376, in get_package_context
    func = getattr(obj, tb.tb_frame.f_code.co_name, "")
  File "/home/user/spack/var/spack/repos/builtin/packages/hdf/package.py", line 133, in libs
    if "+szip" in self.spec:
  File "/home/user/spack/lib/spack/spack/spec.py", line 4193, in __getitem__
    raise KeyError(f"No spec with name {name} in {self}")
KeyError: 'No spec with name zlib in hdf@=4.2.15%gcc@=12.3.0+external-xdr~fortran~java~netcdf+pic~shared~szip build_system=autotools patches=49733dd,5434f29,ce75518,d54592d,f7d67e8 arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] berkeley-db@=18.1.40%gcc@=8.5.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-oracle8-skylake_avx512 ^[deptypes=build] bison@=3.8.2%gcc@=12.3.0~color build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] bzip2@=1.0.8%gcc@=8.5.0~debug~pic+shared build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build] ca-certificates-mozilla@=2023-05-30%gcc@=8.5.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build] cmake@=3.27.9%gcc@=12.3.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] curl@=8.7.1%gcc@=8.5.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-oracle8-skylake_avx512 ^[deptypes=build] diffutils@=3.10%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] findutils@=4.9.0%gcc@=8.5.0 build_system=autotools patches=440b954 arch=linux-oracle8-skylake_avx512 ^[deptypes=build] flex@=2.6.3%gcc@=12.3.0+lex~nls build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=link] gcc-runtime@=12.3.0%gcc@=12.3.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=link] gcc-runtime@=8.5.0%gcc@=8.5.0 build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gdbm@=1.23%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gettext@=0.22.5%gcc@=8.5.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=iconv,libc] glibc@=2.28%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] gmake@=4.4.1%gcc@=8.5.0~guile build_system=generic arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] krb5@=1.20.1%gcc@=12.3.0+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=jpeg] libjpeg-turbo@=3.0.0%gcc@=12.3.0~ipo~jpeg8~partial_decoder+pic build_system=cmake build_type=Release generator=make libs=shared,static arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] libsigsegv@=2.14%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=rpc] libtirpc@=1.3.3%gcc@=12.3.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] libxml2@=2.10.3%gcc@=8.5.0+pic~python+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] m4@=1.4.19%gcc@=8.5.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-oracle8-skylake_avx512 ^[deptypes=build] nasm@=2.15.05%gcc@=12.3.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] ncurses@=6.5%gcc@=8.5.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] nghttp2@=1.57.0%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] openssl@=3.3.0%gcc@=8.5.0~docs+shared build_system=generic certs=mozilla arch=linux-oracle8-skylake_avx512 ^[deptypes=build,run] perl@=5.38.0%gcc@=8.5.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-oracle8-skylake_avx512 ^[deptypes=run] pigz@=2.8%gcc@=8.5.0 build_system=makefile arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run virtuals=pkgconfig] pkgconf@=2.2.0%gcc@=8.5.0 build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] readline@=8.2%gcc@=8.5.0 build_system=autotools patches=bbf97f1 arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link] tar@=1.34%gcc@=8.5.0 build_system=autotools zip=pigz arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] xz@=5.4.6%gcc@=8.5.0~pic build_system=autotools libs=shared,static arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link virtuals=zlib-api] zlib-ng@=2.1.6%gcc@=8.5.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-oracle8-skylake_avx512 ^[deptypes=build,link,run] zstd@=1.5.6%gcc@=8.5.0+programs build_system=makefile compression=none libs=shared,static arch=linux-oracle8-skylake_avx512'
==> Error: Failed to install grads due to InstallError: The process has stopped unexpectedly (exit 1)
